File: fonte/u00 Nivelamento/c/recursividade/fibonaccirecursivo.c
#include <stdio.h> 
#include <stdlib.h> 

int fibonacci(int n){
   int resp;
   printf("\nfib (%i)", n);
   resp = (n == 0 || n == 1) ? 1 : fibonacci(n-1) + fibonacci(n-2);
   printf("\nfib n(%i): %i", n, resp);
   return resp;
}


int main(int argc, char *argv[]) { 
   int n = 5;
   printf("\nFIBONACCI RECURSIVO(%i): %i", n, fibonacci(n));

   return 0; 
}

File: fonte/u00 Nivelamento/java/recursividade/FibonacciRecursivo.java
/**
 * Recursividade
 * @author Max do Val Machado
 * @version 2 01/2015
 */
class FibonacciRecursivo {
   public static int fibRec (int n){
      int resp;
      System.out.println("fib (" + n + ")");
      resp = (n == 0 || n == 1) ? 1 : fibRec(n-1) + fibRec(n-2);
      System.out.println("fib (" + n + "): " + resp);
      return resp;
   }
   public static void main (String[] args){
      int n = Integer.parseInt(args[0]);
      System.out.println("FIBONACCI RECURSIVO(" + n + "): " + fibRec(n));
   }
}
